src/Microsoft.CodeAnalysis.Razor/DefaultTagHelperDescriptorFactory.cs

```diff
@@ -429,41 +393,34 @@ private IEnumerable<IPropertySymbol> GetAccessibleProperties(INamedTypeSymbol ty
             return accessibleProperties.Values;
         }
 
-        private bool ShouldSkipDescriptorCreation(ISymbol symbol)
+        private void AddVisibility(ISymbol symbol, TagHelperDescriptorBuilder builder)
         {
-            if (DesignTime)
+            var editorBrowsableAttribute = symbol.GetAttributes().Where(a => a.AttributeClass == _editorBrowsableAttributeSymbol).FirstOrDefault();
+            if (editorBrowsableAttribute == null)
             {
-                var editorBrowsableAttribute = symbol.GetAttributes().Where(a => a.AttributeClass == _editorBrowsableAttributeSymbol).FirstOrDefault();
-
-                if (editorBrowsableAttribute == null)
-                {
-                    return false;
-                }
-
-                if (editorBrowsableAttribute.ConstructorArguments.Length > 0)
-                {
-                    return (EditorBrowsableState)editorBrowsableAttribute.ConstructorArguments[0].Value == EditorBrowsableState.Never;
-                }
+                return;
             }
 
-            return false;
+            if (editorBrowsableAttribute.ConstructorArguments.Length > 0 &&
+                (EditorBrowsableState)editorBrowsableAttribute.ConstructorArguments[0].Value == EditorBrowsableState.Never)
+            {
+                builder.AddMetadata("EditorBrowsable", "Never");
+            }
         }
 
-        /// <summary>
-        /// Converts from pascal/camel case to lower kebab-case.
-        /// </summary>
-        /// <example>
-        /// SomeThing => some-thing
-        /// capsONInside => caps-on-inside
-        /// CAPSOnOUTSIDE => caps-on-outside
-        /// ALLCAPS => allcaps
-        /// One1Two2Three3 => one1-two2-three3
-        /// ONE1TWO2THREE3 => one1two2three3
-        /// First_Second_ThirdHi => first_second_third-hi
-        /// </example>
-        internal static string ToHtmlCase(string name)
+        private void AddVisibility(ISymbol symbol, ITagHelperBoundAttributeDescriptorBuilder builder)
         {
-            return HtmlCaseRegex.Replace(name, HtmlCaseRegexReplacement).ToLowerInvariant();
+            var editorBrowsableAttribute = symbol.GetAttributes().Where(a => a.AttributeClass == _editorBrowsableAttributeSymbol).FirstOrDefault();
+            if (editorBrowsableAttribute == null)
+            {
+                return;
+            }
+
+            if (editorBrowsableAttribute.ConstructorArguments.Length > 0 &&
+                (EditorBrowsableState)editorBrowsableAttribute.ConstructorArguments[0].Value == EditorBrowsableState.Never)
+            {
+                builder.AddMetadata("EditorBrowsable", "Never");
```

Comment:
Yeah I wasn't sure if this should be first class on the API or just a metadata, metadata feels good.